### PR TITLE
Replace deprecated bot.privileges accesses

### PR DESCRIPTION
**Pull request by half-duplex**
`bot.privileges[channel][user]` -> `bot.channels[channel].privileges[user]`. Smoke tested.

### CODE REVIEW

**Comment by dgw** [comment]
Squash please!

Also, your PR title is much clearer than the commit message. We're not replacing `bot.privileges`, but only uses of it in core code.

**Comment by dgw** [comment]
Now that I think about it, if this had been done way back when `bot.privileges` was first deprecated, I bet `bot.channels` never would have been released broken the way it was.

We should definitely be using the "new and improved" things internally, not the old deprecated stuff. This should have happened ages ago.

File: sopel/test_tools.py [hunk 1]
```diff
@@ -47,7 +47,9 @@ def __init__(self, nick, admin=False, owner=False):
         self.nick = nick
         self.user = "sopel"
 
-        self.channels = ["#channel"]
+        channel = sopel.tools.Identifier("#Sopel")
+        self.channels = {
+            channel: sopel.tools.target.Channel(channel)}
```
Comment (dgw) [inline sopel/test_tools.py]: What's up with this line break? The line wouldn't be that long without it.

Comment (half-duplex) [reply sopel/test_tools.py]: I think I was using a longer variable name than `channel` when I made it.

**Comment by dgw** [comment]
@half-duplex This is approved, but waiting on a hotfix for #1482 (consistency between `bot.privileges` and `bot.channels` data). Without it, this could cause modules to start behaving differently with certain ircds, due to the mismatched mode info, and we don't want to introduce such changes in a patch release.

**Pull request merged**